backend/services/application.py

```diff
@@ -0,0 +1,230 @@
+"""Service that manages applications for the COMP department"""
+
+from typing import List
+from fastapi import Depends
+from sqlalchemy import update, delete
+from sqlalchemy.orm import Session
+from typing import Dict
+from backend.entities import section_application_table
+from backend.entities.application_entity import ApplicationEntity, New_UTA_Entity
+from backend.entities.section_application_table import section_application_table
+from backend.entities.academics.section_entity import SectionEntity
+from backend.models.academics.section import Section
+from backend.models.application_details import (
+    New_UTADetails,
+    UserApplication,
+)
+from backend.models.user import User
+
+from backend.services.exceptions import ResourceNotFoundException
+
+from ..database import db_session
+
+__authors__ = ["Ben Goulet, Abdulaziz Al-Shayef"]
+__copyright__ = "Copyright 2024"
+__license__ = "MIT"
+
+
+class ApplicationService:
+    """ApplicationService is the access layer to TA applications."""
+
+    def __init__(self, session: Session = Depends(db_session)):
+        """Initializes a new ApplicationService.
+
+        Args:
+            session (Session): The database session to use, typically injected by FastAPI.
+        """
+        self._session = session
+
+    def list(self) -> list[New_UTADetails]:
+        """Returns all TA applications.
+
+        Returns:
+            list[New_UTA]: List of all current and previously submitted applications.
+        """
+        entities = self._session.query(ApplicationEntity).all()
+
+        return [entity.to_details_model() for entity in entities]
+
+        # implement list() for all types here later
+
+        # applications = []
+
+        # for entity in entities:
+        #     if entity.type == "new_uta": # type: ignore
+        #         applications.append(
+        #             entity.to_details_model()
+        #         )
+        #     elif entity.type == "returning_uta":
+        #         applications.append(entity.to_details_model())
+        #     elif entity.type == "uta":
+        #         applications.append(entity.to_details_model())
+        #     else:
+        #         applications.append(entity.to_model())
+
+        # return applications
+        
+    
+    def get_application(self, subject: User) -> UserApplication:
+        """Returns application(s) for a specific user
+
+        Returns:
+            list[Application]: List of all current applications for a specific user
+        """
+
+        application_entity = (
+            self._session.query(ApplicationEntity)
+            .filter(ApplicationEntity.user_id == subject.id)
+            .first()
```

Comment:
For this to work, make sure that this is sorted! This might cause issues as users apply in many terms. In addition, there are some edge cases to consider:
* Users may have active applications for say, Summer Session I and Summer Session II at the same time.
* Users might take a break from being a TA and not submit an application for the current term, but they may have applied in the past. Should this function return the past application (i.e., the most recent one), or `None` if they did not apply to an application for currently active terms?